RyuSocks.Test/Auth/UsernameAndPasswordTests.cs

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) RyuSOCKS
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+using RyuSocks.Auth;
+using RyuSocks.Auth.Packets;
+using System;
+using System.Collections.Generic;
+using System.Security.Authentication;
+using Xunit;
+
+namespace RyuSocks.Test.Auth
+{
+    public class UsernameAndPasswordTests
+    {
+        private static readonly UsernameAndPasswordResponse _expectedUsernameAndPasswordResponse = new()
+        {
+            Version = AuthConsts.UsernameAndPasswordVersion,
+            Status = 0,
+        };
+
+        [Theory]
+        [InlineData("Username", "Password", true)]
+        [InlineData("InvalidUsername", "InvalidPassword", false)]
```

Comment:
Add the following test cases:

- valid username, invalid password
- invalid username, valid password